src/resource_format/gdj_resource_format_loader.cpp

```diff
@@ -27,25 +27,29 @@ Error kt_read_all_file_utf8(const String& p_path, String& r_content) {
 }
 
 Ref<Resource> GdjResourceFormatLoader::load(const String& p_path, const String& p_original_path, Error* r_error, bool p_use_sub_threads, float* r_progress, CacheMode p_cache_mode) {
+    // We don't import Kotlin scripts so p_path == p_original_path
+    String script_name = p_path.get_file().trim_suffix(p_path.get_extension()).trim_suffix(".");
+    Ref<KotlinScript> ref = TypeManager::get_instance().get_user_script_from_name(script_name);
+    if (ref.is_null()) {
 #ifdef TOOLS_ENABLED
-    // TODO: check if we need to take CacheMode into account like GDScript does
-    Ref<KotlinScript> ref = KotlinScriptCache::get_or_create_script(p_path);
-#else
-    Ref<KotlinScript> ref {memnew(KotlinScript)};
+        // If we reach that location, it means that the script file being loaded hasn't been built into the .jar.
+        // We create a script placeholder instead. When reloading, it will be properly updated with the correct KtClass.
+        ref = TypeManager::get_instance().create_placeholder_script(p_path);
+#elif DEBUG_ENABLED
+        // All scripts are supposed to be already in cache when not in the editor.
+        if (r_error) { *r_error = Error::ERR_UNAUTHORIZED; }
```

Comment:
Not sure if `ERR_UNAUTHORIZED` is the right error to return here. Why did you chose this one?

Reply:
I just checked the list, and most of them are very specific error that didn't seem to fit the situation. So I just used that one because it's "vague" enough.